lib/crawler_web/controllers/api/v1/user_registration_controller.ex

```diff
@@ -0,0 +1,51 @@
+defmodule CrawlerWeb.Api.V1.UserRegistrationController do
+  use CrawlerWeb, :controller
+
+  alias Crawler.Account.{Accounts, Guardian}
+  alias Crawler.Account.Schemas.User
+  alias CrawlerWeb.Api.ErrorView
+
+  def create(conn, %{"email" => email, "password" => password}) do
+    case Accounts.register_user(%{"email" => email, "password" => password}) do
+      {:ok, user} ->
+        process_user_encode_and_sign(conn, user)
+
+      {:error, _changeset} ->
+        conn
+        |> put_view(ErrorView)
+        |> put_status(:unprocessable_entity)
+        |> render("error.json", %{
+          code: :unprocessable_entity,
+          detail: gettext("%{email} has already been taken", email: email)
```

Comment:
We need to display changeset errors here, instead of custom text.

It is possible that password will not pass validation, but you will show to the use message that email was already been taken: https://github.com/vnntsu/elixir-ic/blob/cbbf48f3290af8522e858c49e2d5af17c850c3a9/lib/crawler/account/schemas/user.ex#L65

Reply:
@Nihisil Fixed in [6b60203](https://github.com/vnntsu/elixir-ic/pull/45/commits/6b6020300ddcf556432cea98dd5496627c73a502)